packages/ui-components/src/components/input/fileAvatarInput.tsx

```diff
@@ -0,0 +1,143 @@
+  const [preview, setPreview] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const onDrop = useCallback(
+    (acceptedFiles: Array<File>, onDropRejected) => {
+      if (onDropRejected.length !== 0) {
+        onError(onDropRejected[0].errors[0]);
+      } else {
+        /**
+         * Render image to calculate the dimensions
+         */
+        setLoading(true);
+        const image: HTMLImageElement = new Image();
+        image.addEventListener('load', () => {
+          setLoading(false);
+          if (
+            image.width > maxDimension ||
+            image.height > maxDimension ||
+            image.width < minDimension ||
+            image.height < minDimension ||
+            image.height !== image.width // check if the image is squere or not
+          ) {
+            onError({
+              code: 'wrong-dimension',
+              message:
+                'Please provide a squared image with size between 256px and 2400 px on each side',
+            });
+          } else {
+            onChange(acceptedFiles[0]);
+            setPreview(image.src);
+          }
+        });
+        image.src = URL.createObjectURL(acceptedFiles[0]);
+      }
+    },
+    [maxDimension, minDimension, onChange, onError]
+  );
+
+  const {getRootProps, getInputProps} = useDropzone({
+    onDrop,
+    maxSize: maxFileSize,
+    accept: 'image/jpg, image/jpeg, image/png, image/gif, image/svg',
+  });
+  return (
+    <>
+      {loading ? (
+        <LoadingContainer>
+          <Spinner size="small" />
+        </LoadingContainer>
+      ) : (
+        <>
+          {preview ? (
+            <ImageContainer>
+              <Preview src={preview} />
+              <StyledButton
+                icon={<IconClose />}
+                size="small"
+                mode="secondary"
+                onClick={() => {
+                  setPreview(null);
+                  onChange(null);
+                }}
+              />
+            </ImageContainer>
+          ) : (
+            <DefaultContainer
+              data-testid="fileAvatar-input"
+              {...getRootProps()}
+            >
+              <IconAdd />
+              <input {...getInputProps()} />
+            </DefaultContainer>
+          )}
+        </>
+      )}
+    </>
+  );
+};
+
+const DefaultContainer = styled.div.attrs({
+  className: `flex items-center justify-center bg-ui-0 text-ui-600 
+  h-8 w-8 border-dashed border-ui-100 border-2 rounded-xl cursor-pointer`,
+})``;
+
+const LoadingContainer = styled.div.attrs({
+  className: `flex items-center justify-center bg-ui-0 
+    h-8 w-8 border-dashed border-primary-500 border-2 rounded-xl`,
+})``;
+
+const ImageContainer = styled.div.attrs({
+  className: 'relative h-8 w-8',
+})``;
+
+const Preview = styled.img.attrs({
+  className: 'rounded-xl bg-ui-0 h-8 w-8',
+})``;
+
+const StyledButton = styled(ButtonIcon).attrs({
+  className: 'absolute',
+})`
+  top: -10px;

```

Comment:
```suggestion
  className: 'absolute -top-2 -right-1.75 ',
```
And then you can delete `top` and `left` on line 139 and 140